transport_test.go

```diff
@@ -42,3 +57,143 @@ func TestInitTransportWithCustomCA(t *testing.T) {
 		t.Error("expected root CA to be set, got nil")
 	}
 }
+
+type testHTTPHandler struct {
```

Comment:
Could we use [httptest.Server](https://pkg.go.dev/net/http/httptest#Server) instead?

Reply:
I did not find a way how to modify its TLS configuration properly. Even if we did that, you still need to supply your own handler.